### PR TITLE
PLANNER-1027:  Fixed benchmark

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

File: optashift-employee-rostering-benchmark/src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -2,16 +2,26 @@
 <!-- OVERWRITES the normal persistence.xml in the server module -->
 <persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              version="2.1" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
-  <persistence-unit name="optashift-employee-rostering-persistence-unit">
+  <persistence-unit name="optashift-employee-rostering-persistence-unit" transaction-type="RESOURCE_LOCAL">
```
Comment (ge0ffrey) [inline optashift-employee-rostering-benchmark/src/main/resources/META-INF/persistence.xml]: This resource local might be the reason you're getting empty lists.
IIRC, resource local means you're say you will explicitly open and close all transactions everywhere in your code.
So the generated testdata might not get commited to the database.

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java [hunk 2]
```diff
@@ -93,6 +95,7 @@ public RosterGenerator() {}
      */
     public RosterGenerator(EntityManager entityManager) {
         this.entityManager = entityManager;
+        shiftRestService = new ShiftRestServiceImpl(entityManager);
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java]: The generator should probably not talk to any rest service, if possible.

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/shift/ShiftRestServiceImpl.java [hunk 3]
```diff
@@ -64,6 +65,17 @@
     @Inject
     private EmployeeRestService employeeRestService;
 
+    @SuppressWarnings("unused")
+    public ShiftRestServiceImpl() {
+    }
+
+    // For benchmark
+    public ShiftRestServiceImpl(EntityManager entityManager) {
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/shift/ShiftRestServiceImpl.java]: I 'd rather see the Generator refactor so it doesn't need the ShiftRestService to begin with.

File: optashift-employee-rostering-benchmark/src/main/resources/META-INF/persistence.xml (hunk 1 above)
Comment (Christopher-Chianelli) [reply optashift-employee-rostering-benchmark/src/main/resources/META-INF/persistence.xml]: Changing it to JTA causes a NPE when building the entityManager (no other info is given).

Comment (rsynek) [reply optashift-employee-rostering-benchmark/src/main/resources/META-INF/persistence.xml]: @Christopher-Chianelli for JTA transactions, I would suggest configuring a Datasource on the container - I can't see any used in your persistence.xml

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java [hunk 4]
```diff
@@ -91,8 +90,9 @@ public RosterGenerator() {
      * For benchmark only
      * @param entityManager never null
      */
-    public RosterGenerator(EntityManager entityManager) {
+    public RosterGenerator(EntityManager entityManager, ShiftFileParser shiftGenerator) {
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java]: What is ShiftFileParser? What does it do.

The variable name called generator but the type is FileParser - that's confusing.

Comment (Christopher-Chianelli) [reply optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java]: ShiftFileParser probably should be renamed to `ShiftGenerator`. It converts `ShiftTemplate` to `Shift`.

Comment (Christopher-Chianelli) [reply optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java]: (FileParser part is from when it parses files in the October 2017 version)

Comment (ge0ffrey) [reply optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java]: Now that ShiftTemplate with have a createShift(start, end) method, is Shiftgenerator/ShiftTemplate still needed?

**Comment by yurloc** [comment]
Are you sure this is the fix? Why is test coverage missing? How is `OptaShiftEmployeeRosteringBenchmarkApp` supposed to be used? It still fails if I run
```
mvn -nsu exec:java -Dexec.mainClass=org.optaplanner.openshift.employeerostering.benchmark.OptaShiftEmployeeRosteringBenchmarkApp
```

**Comment by yurloc** [comment]
I'm sorry I messed up the local build for the first time. Now the benchmark runs OK.

Only the missing test coverage remark is still valid. If the benchmark breaks in the future we won't notice it without at least a basic test.

**Pull request merged**